apps/seqan_flexbar/general_processing.h

```diff
@@ -483,474 +483,240 @@ void processN(TSeqs& seqs, TIds& ids, TSeqs& seqsRev, TIds& idsRev, TMulti& mult
+	int limit = length(seqs);
 
-    resize(rem, limit);
-    if (head > 0 && tail > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) > (head + tail))
-            {
-                erase(seqs[i], 0, head);
-                erase(seqs[i], length(seqs[i]) - tail, length(seqs[i]));
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (head > 0)
-    {
-       SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-           if (length(seqs[i]) > head)
-            {
-                erase(seqs[i], 0, head);
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else if (tail > 0)
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) > tail)
-            {
-                erase(seqs[i], length(seqs[i]) - tail, length(seqs[i]));
-                if (length(seqs[i]) >= min)
-                {
-                    rem[i] = false;
-                }
-                else
-                {
-                    rem[i] = true;
-                }
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    else
-    {
-        SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
-        for (i = 0; i < limit; ++i)
-        {
-            if (length(seqs[i]) >= min)
-            {
-                rem[i] = false;
-            }
-            else
-            {
-                rem[i] = true;
-            }
-        }
-    }
-    unsigned ex = 0;
-    for (int j = length(rem) - 1; j >= 0; --j)
-    {
-        if (rem[j])
-        {
-            swap(seqs[j], seqs[limit - ex -1]);
-            swap(ids[j], ids[limit - ex -1]);
-            swap(multiplex[j], multiplex[limit - ex -1]);
-            ++ex;
-        }
-    }
-    if (ex != 0)
-    {
-        resize(seqs, limit - ex);
-        resize(ids, limit - ex);
-        resize(multiplex, limit - ex);
-        stats.removedSeqsShort += ex;
-    }
+	resize(rem, limit);
+
+	// disable this omp, because making a local copy of ids for every threat is not worth it
+	//SEQAN_OMP_PRAGMA(parallel for default(shared) private(i) schedule(static))
+	for (i = 0; i < limit; i++)
```

Comment:
hm yes thats true. I dont know the internal design of the underlaying structure, if its designed so that different threads can work on different elements at the same time, then it should be ok. I guess deleting or inserting would be problematic, but changing the content is ok right?